src/actions/palette.js

```diff
@@ -6,3 +8,31 @@ export function setPaletteColor(name, colorCode) {
     payload: { name, colorCode },
   };
 }
+
+export function changePaletteColor(name, colorCode) {
+  return (dispatch) => {
+    const linkContainColorName = doesContinParam(name);
+    const separator = window.location.href.includes('?') ? '&' : '?';
+    const newLink = linkContainColorName
+      ? replaceUrlDataWith(name, colorCode)
+      : `${window.location.href + separator + name}=${colorCode}`;
+    window.history.pushState('', '', newLink);
+    dispatch(setPaletteColor(name, colorCode));
+  };
+}
+
+export function setInitialPaletteColors() {
+  return (dispatch) => {
+    const params = getParamsFromUrl(window.location.href);
+    if (!params) {
+      return;
+    }
+    // TODO: filter params array from non color params
+    // but now work without this
+    params.forEach((param) => {
+      const colorName = param.split('=')[0];
+      const colorCode = param.split('=')[1];
+      dispatch(setPaletteColor(colorName, colorCode));
```

Comment:
rather than calling dispatch in a loop, i prefer calling dispatch once providing an array of new values